### DISCUSSION TITLE
Can I create instance of Image struct with Rg32float texture?

### DISCUSSION THREAD

**Discussion by vino-volare** [bevyengine/bevy#13363] [Q&A] [answered]
Hi! I'm learning  how to work with custom material in bevy.
I want to send `Rg32float` texture by instance of `Image` struct to fragment shader, But I think that can't create instance of `Image` struct with `Rg32float` texture because type of `Image.data` is `Vec<u8>`not `Vec<f32>`.
But `Rg32float` is exist in `TextureFormat`, the type of `Image.format`.
Can I ignore that difference of type?

**Comment by JMS55** [comment, marked as answer]
For one pixel of an Rg32Float texture, it'll be 8 bytes. 4 bytes for the R channel, and another 4 bytes for the G channel. So you should convert your [f32; 2] into [u8; 8]. You can use crates like bytemuck to perform the cast.

**Reply by vino-volare** [reply]
Thanks for answering! I understood. I'll try this.:)
